flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/SqlClient.java

```diff
@@ -121,6 +132,114 @@ private void start() {
         }
     }
 
+    private List<URL> discoverDependencies(List<URL> jars, List<URL> libraries) {
```

Comment:
can we move these new methods into a utility class, this class has been too complex.